File: crossecore-csharp/Java/Java/Java_/PostfixExpressionBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class PostfixExpressionBase 
	:ExpressionImpl, PostfixExpression
	{
		private PostfixExpressionKind _operator = PostfixExpressionKind.INCREMENT;
		public virtual PostfixExpressionKind operator_
		{
		get { 
			return _operator;
		}
		set { _operator = value; }
		}
		private Expression _operand;
		public virtual Expression operand
		{
			get {
			
				return _operand;
			}
			set {
				if (value != _operand) {
					NotificationChain msgs = null;
					if (_operand != null){
						msgs = ((InternalEObject)_operand).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.POSTFIXEXPRESSION_OPERAND, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.POSTFIXEXPRESSION_OPERAND, null, msgs);
					}
					msgs = basicSetOperand(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.POSTFIXEXPRESSION_OPERAND , value, value));
				}
				}
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.POSTFIXEXPRESSION;
		}
		
		
		public NotificationChain basicSetOperand(Expression newobj, NotificationChain msgs) {
			var oldobj = _operand;
			_operand = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.POSTFIXEXPRESSION_OPERAND, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		
		public override Object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.POSTFIXEXPRESSION_COMMENTS:
					return comments;
				case Java_PackageImpl.POSTFIXEXPRESSION_ORIGINALCOMPILATIONUNIT:
					return originalCompilationUnit;
				case Java_PackageImpl.POSTFIXEXPRESSION_ORIGINALCLASSFILE:
					return originalClassFile;
				case Java_PackageImpl.POSTFIXEXPRESSION_OPERATOR:
					return operator_;
				case Java_PackageImpl.POSTFIXEXPRESSION_OPERAND:
					return operand;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.POSTFIXEXPRESSION_COMMENTS:
					comments.Clear();
					comments.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.POSTFIXEXPRESSION_ORIGINALCOMPILATIONUNIT:
					originalCompilationUnit = (CompilationUnit) newValue;
					return;
				case Java_PackageImpl.POSTFIXEXPRESSION_ORIGINALCLASSFILE:
					originalClassFile = (ClassFile) newValue;
					return;
				case Java_PackageImpl.POSTFIXEXPRESSION_OPERATOR:
					operator_ = (PostfixExpressionKind) newValue;
					return;
				case Java_PackageImpl.POSTFIXEXPRESSION_OPERAND:
					operand = (Expression) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		
	}
}

File: crossecore-typescript/java_/PostfixExpressionBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {Bag} from "ecore/Bag";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ExpressionImpl} from "./ExpressionImpl";
import {PostfixExpressionKind} from "./PostfixExpressionKind";
import {Expression} from "./Expression";
import {PostfixExpression} from "./PostfixExpression";
		
			export class PostfixExpressionBase
			extends ExpressionImpl
			implements PostfixExpression
			{
				private _operator:PostfixExpressionKind = PostfixExpressionKind.INCREMENT;
				get operator():PostfixExpressionKind{
					return this._operator;
				}
				set operator(value:PostfixExpressionKind){
					this._operator = value; 
				}
				private _operand:Expression = null;
				get operand():Expression{
				
					return this._operand;
				}
				set operand(value:Expression) {
					if (value != this._operand) {
						let msgs:NotificationChain = null;
						if (this._operand != null){
							msgs = (this._operand).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.POSTFIXEXPRESSION_OPERAND, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.POSTFIXEXPRESSION_OPERAND, /*null*/ null, msgs);
						}
						msgs = this.basicSetOperand(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.POSTFIXEXPRESSION_OPERAND , value, value));
					}
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return PostfixExpressionBase.eStaticClass;
				}
			
			
				public basicSetOperand(newobj:Expression, msgs:NotificationChain):NotificationChain {
					let oldobj = this._operand;
					this._operand = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.POSTFIXEXPRESSION_OPERAND, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.POSTFIXEXPRESSION_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.POSTFIXEXPRESSION_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.POSTFIXEXPRESSION_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.POSTFIXEXPRESSION_OPERATOR:
							return this.operator;
						case Java_PackageLiterals.POSTFIXEXPRESSION_OPERAND:
							return this.operand;
					}
					//return this.eGetFromExpression(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.POSTFIXEXPRESSION_OPERATOR:
							this.operator = <PostfixExpressionKind> newValue;
							return;
						case Java_PackageLiterals.POSTFIXEXPRESSION_OPERAND:
							this.operand = <Expression> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
